pkg/api/types.go

```diff
@@ -421,6 +425,19 @@ type NFSVolumeSource struct {
 	ReadOnly bool `json:"readOnly,omitempty"`
 }
 
+// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the lifetime of a pod
+type GlusterfsVolumeSource struct {
+	// Required: EndpointsName is the endpoint name that details Glusterfs topology
+	EndpointsName string `json:"endpoints"`
```

Comment:
It's not unreasonable to force someone to create an external headless service if they want this behavior.

----- Original Message -----
> > @@ -421,6 +425,19 @@ type NFSVolumeSource struct {
> >  	ReadOnly bool `json:"readOnly,omitempty"`
> >  }
> >  
> > +// GlusterfsVolumeSource represents a Glusterfs Mount that lasts the
> > lifetime of a pod
> > +type GlusterfsVolumeSource struct {
> > +	// Required: EndpointsName is the endpoint name that details Glusterfs
> > topology
> > +	EndpointsName string `json:"endpoints"`
> 
> Is the assumption that endpoints for gluster lie outside the kubernetes
> cluster?  I am a bit anxious about direct creation of endpoints without a
> service (now that we have headless services) since it lays a trap for a
> later collision that won't be detected.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/6174/files#r27942357